src/newgrf.cpp

```diff
@@ -9068,20 +9068,13 @@ static void CalculateRefitMasks()
 		 * cargo type. Finally disable the vehicle, if there is still no cargo. */
 		if (!IsValidCargoID(ei->cargo_type) && ei->refit_mask != 0) {
 			/* Figure out which CTT to use for the default cargo, if it is 'first refittable'. */
-			const uint8_t *cargo_map_for_first_refittable = nullptr;
-			{
-				const GRFFile *file = _gted[engine].defaultcargo_grf;
-				if (file == nullptr) file = e->GetGRF();
-				if (file != nullptr && file->grf_version >= 8 && file->cargo_list.size() != 0) {
-					cargo_map_for_first_refittable = file->cargo_map;
-				}
-			}
-
-			if (cargo_map_for_first_refittable != nullptr) {
+			const GRFFile *file = _gted[engine].defaultcargo_grf;
+			if (file == nullptr) file = e->GetGRF();
+			if (file != nullptr && file->grf_version >= 8 && file->cargo_list.size() != 0) {
```

Comment:
```suggestion
			if (file != nullptr && file->grf_version >= 8 && !file->cargo_list.empty()) {
```
It's already existing code that got re-indented. So, it might also be something for a later PR that looks for other `size() == 0` vs `size() != 0` cases.